jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -1771,14 +1791,25 @@ public boolean equalsIgnoreCase(String string) {
 
 			if (charAtO1Last != charAtO2Last &&
 					toUpperCase(charAtO1Last) != toUpperCase(charAtO2Last) &&
-					((charAtO1Last <= 255 && charAtO2Last <= 255) || Character.toLowerCase(charAtO1Last) != Character.toLowerCase(charAtO2Last))) {
+					((charAtO1Last <= 255 && charAtO2Last <= 255) || Character.toLowerCase(charAtO1Last) != Character.toLowerCase(charAtO2Last)) &&
+					(!Character.isLowSurrogate(charAtO1Last) || !Character.isLowSurrogate(charAtO2Last))) {
```

Comment:
I think this should be included for Java16+ only.

Reply:
I think there is a problem here for Java 16+. If the last character is a low surrogate, then it's not checked for equality here. If the character proceeding it isn't a high surrogate, true can be returned even when the characters are different.

Reply:
Pls create a test case for this case, which should fail with this code, and pass once the problem is fixed.

Reply:
Updated method and included a test with low surrogate half only at the end of a string/region.